libs/dkg/include/dkg/rbc_messages.hpp

```diff
@@ -46,14 +46,27 @@ class RBCMessage
 public:
   enum class MessageType : uint8_t
   {
-    RBROADCAST,
-    RECHO,
-    RREADY,
-    RREQUEST,
-    RANSWER
+    R_BROADCAST,
+    R_ECHO,
+    R_READY,
+    R_REQUEST,
+    R_ANSWER,
+    R_SEND
   };
 
   // Destruction
+  RBCMessage() = default;
+  RBCMessage(uint16_t channel, uint32_t id, uint8_t counter, SerialisedMessage msg = "")
+    : channel_{channel}
+    , id_{id}
+    , counter_{counter}
+    , payload_(std::move(msg))
+  {}
+  explicit RBCMessage(RBCSerializer &serialiser)
+  {
+    serialiser >> channel_ >> id_ >> counter_ >> payload_;
+  }
+
```

Comment:
I don't think the RBCMessage constructors should be public. One should not be able to create a message without specifying a type for it.

Reply:
Yes, I agree, but this will first be changed in subsequent PRs as this is just the first in many minor changes.